src/test/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -0,0 +1,371 @@
+        testCases.add(new ShouldSkipTripTestCase(
+            "should return true if trip is starting in 45 minutes and the last time checked was 2 minutes ago",
+            exemplarDateTime.withHour(7).withMinute(55), // 7:55am
+            true,
+            laterTodayTrip4
+        ));
+
+        // - Return false if trip is starting in 10 minutes and the last time checked was 2 minutes ago
+        // The monitored trip lead time is 30 minutes, so check every minute
+        MonitoredTrip laterTodayTrip5 = createMonitoredTrip(user.id, otpDispatcherResponse);
+        laterTodayTrip5.updateWeekdays(true);
+        // last checked at 8:23am
+        setLastCheckedTimeForTripJourneyState(laterTodayTrip5, exemplarDateTime.withHour(8).withMinute(28));
+        testCases.add(new ShouldSkipTripTestCase(
+            "should return false if trip is starting in 10 minutes and the last time checked was 2 minutes ago",
+            exemplarDateTime.withHour(8).withMinute(30), // 8:30am
+            false,
+            laterTodayTrip5
+        ));
+
+        // - Return true if trip has started 3 minutes ago
+        MonitoredTrip tripThatJustStarted = createMonitoredTrip(user.id, otpDispatcherResponse);
+        tripThatJustStarted.updateWeekdays(true);
+        // last checked at 8:39am
+        setLastCheckedTimeForTripJourneyState(tripThatJustStarted, exemplarDateTime.withHour(8).withMinute(39));
+        testCases.add(new ShouldSkipTripTestCase(
+            "should return true if trip has started 3 minutes ago",
+            exemplarDateTime.withHour(8).withMinute(43), // 8:43am
+            true,
+            tripThatJustStarted
+        ));
+
+        return testCases;
+    }
+
+    private static void setLastCheckedTimeForTripJourneyState(MonitoredTrip trip, ZonedDateTime dateTime) {
+        JourneyState journeyState = trip.retrieveJourneyState();
+        journeyState.lastChecked = dateTime.toInstant().toEpochMilli();
+        Persistence.journeyStates.replace(journeyState.id, journeyState);
+    }
+
+    private static class ShouldSkipTripTestCase {
+        /* a helpful message describing the particular test case */
+        public final String message;
+        /* The time to mock for this test case */
+        public final ZonedDateTime mockTime;
+        /**
+         * if true, it is expected that the {@link CheckMonitoredTripTest#createSkipTripTestCases()} method should
+         * calculate that the given trip should be skipped.
+         */
+        public final boolean shouldSkipTrip;
+        /**
+         * The trip for the {@link CheckMonitoredTripTest#createSkipTripTestCases()} method to calculate whether
+         * skipping trip analysis should occur.
+         */
+        public final MonitoredTrip trip;
+
+        private ShouldSkipTripTestCase(
+            String message, ZonedDateTime mockTime, boolean shouldSkipTrip, MonitoredTrip trip
```

Comment:
Might be helpful if the constructor accepts the lastCheckedTime as well. Having that time next to the mocktime might make things more readable.

Reply:
Refactored.